lib/Digitick/Sepa/TransferFile/Factory/TransferFileFacadeFactory.php

```diff
@@ -60,9 +69,18 @@ public static function createDirectDebit($uniqueMessageIdentification, $initiati
     public static function createCustomerCredit($uniqueMessageIdentification, $initiatingPartyName, $painFormat = 'pain.001.002.03')
     {
         $groupHeader = new GroupHeader($uniqueMessageIdentification, $initiatingPartyName);
-        $directDebitTransferFile = new CustomerCreditTransferFile($groupHeader);
-        $domBuilder = new CustomerCreditTransferDomBuilder($painFormat);
 
-        return new CustomerCreditFacade($directDebitTransferFile, $domBuilder);
+        return new CustomerCreditFacade(new CustomerCreditTransferFile($groupHeader), new CustomerCreditTransferDomBuilder($painFormat));
+    }
+
+    /**
+     * @param object $groupHeader
+     * @param string $painFormat
+     *
+     * @return CustomerCreditFacade
+     */
+    public static function createCustomerCreditFromGroupHeader(GroupHeader $groupHeader, $painFormat = 'pain.001.002.03')
```

Comment:
I think it would be better to call the method createCustomer(Credit|Debit)**With**GroupHeader, there is no information in the header used to create the Facade except the header itself